app.c

```diff
@@ -459,28 +555,48 @@ static VOID FinalizeResults(EFI_FILE_PROTOCOL *Csv)
 	Assert(Status == EFI_SUCCESS);
 
 	/* Platform product name */
-	Status = LibGetSystemConfigurationTable (&SMBIOS3TableGuid,
-	                                         (VOID **)&SmbiosTable);
-	Len = UnicodeSPrint(LStr, 100, L"ProductName, \"%a\"\n", GetProductName());
-	Char16toChar8(Str, LStr, Len);
+	Len = AsciiSPrint(Str, 100, "ProductName,\"%a\"\n", GetProductName());
 
 	Status = uefi_call_wrapper(Csv->Write, 3, Csv, &Len, Str);
 	Assert(Status == EFI_SUCCESS);
 
+	/* Store information about populated memory */
+	StoreDimmsInfo(Csv);
+
+	/* Flush before allowing users to do something unexpected */
+	Status = uefi_call_wrapper(Csv->Flush, 1, Csv);
+	Assert(Status == EFI_SUCCESS);
+
+	Print(L"%H");
 	/* Prompt and save temperature */
 	Input(L"Ambient temperature: ", InStr, 10);
 	Print(L"\n");
-	Len = UnicodeSPrint(LStr, 100, L"Temperature, \"%s\"\n", InStr);
-	Char16toChar8(Str, LStr, Len);
+	Len = AsciiSPrint(Str, 100, "Temperature,\"%s\"\n", InStr);
 
 	Status = uefi_call_wrapper(Csv->Write, 3, Csv, &Len, Str);
 	Assert(Status == EFI_SUCCESS);
 
+	/* Flush before allowing users to do something unexpected */
+	Status = uefi_call_wrapper(Csv->Flush, 1, Csv);
+	Assert(Status == EFI_SUCCESS);
+
 	/* Prompt and save power-off time */
 	Input(L"Time (in seconds) without power: ", InStr, 10);
 	Print(L"\n");
-	Len = UnicodeSPrint(LStr, 100, L"Time, \"%s\"\n", InStr);
-	Char16toChar8(Str, LStr, Len);
+	Len = AsciiSPrint(Str, 100, "Time,\"%s\"\n", InStr);
+
+	Status = uefi_call_wrapper(Csv->Write, 3, Csv, &Len, Str);
+	Assert(Status == EFI_SUCCESS);
+
+	/* Flush before allowing users to do something unexpected */
+	Status = uefi_call_wrapper(Csv->Flush, 1, Csv);
+	Assert(Status == EFI_SUCCESS);
+
+	/* Prompt for other comments, 96 + 2 * '"' + '\n' + '\0' = 100 */
+	Input(L"Comments (max 96 characters, leave empty to skip): ", LStr, 96);
```

Comment:
FIXME: either this reads only 48 characters, or `AsciiSPrint` below writes only 50. Most likely the latter, as there is no quotation mark in the output.

Reply:
https://github.com/ncroxon/gnu-efi/issues/63, https://github.com/Dasharo/ram-remanence-tester/pull/5/commits/3d9b174fbefb8c4b0fc28ffd328826e5a284878c